ii-ix/scallybot
# [Feature/Enhancement] Merge 1.2.0 into Main

## Description

This pull request aims to merge the `feature/docker-integration` branch into the `main` branch, incorporating several new features and improvements into the codebase. Here's an overview of the changes included:

### Features and Enhancements:

1. **Docker:**
   - Containerized project to now run in a Docker container instead of an OS Node.js process.

2. **Logs:**
   - Allows user the ability to provide a specific file location for Discord log files.

### Checklist

Please review the following checklist before merging this PR:

- [ ] Code has been reviewed by peers.
- [ ] Unit tests have been updated and pass successfully.
- [ ] Documentation has been updated, including the `README.md`.
- [ ] VS Code customization files have been provided for an improved development experience.
- [ ] Prefix and application command files are ready for further development or are functional as needed.
- [ ] Code follows best practices and style guidelines.
- [ ] No breaking changes have been introduced.

### Related Issues

- Closes #[issue_number] (if applicable, replace [issue_number] with the actual issue number)

## Testing Instructions

Running `docker compose build && docker compose up -d` should successfully run this project in a Docker container named "___scallybot___".

## Reviewers

* [ii-ix (Two Nine)](https://github.com/ii-ix/)

## Checklist for Reviewers

- [ ] Code logic and functionality have been reviewed thoroughly.
- [ ] Code style and formatting adhere to project standards.
- [ ] Documentation changes have been reviewed for accuracy and clarity.
- [ ] Tests have been executed successfully, and edge cases have been considered.
- [ ] Any questions or concerns have been addressed in the comments.
